orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -58,16 +60,15 @@ private static CloseableHttpClient httpClientWithServiceUnavailableRetryStrategy
         @Override
         public boolean retryRequest(HttpResponse response, int executionCount, HttpContext context) {
           int statusCode = response.getStatusLine().getStatusCode();
-          HttpUriRequest currentReq = (HttpUriRequest) context.getAttribute(
-            HttpCoreContext.HTTP_REQUEST);
-
+          HttpUriRequest currentReq = (HttpUriRequest) context.getAttribute(HttpCoreContext.HTTP_REQUEST);
           LOGGER.info("Response code {} for {}", statusCode, currentReq.getURI());
           boolean shouldRetry = (statusCode == 429 || RETRYABLE_500_HTTP_STATUS_CODES.contains(statusCode)) && executionCount <= MAX_RETRIES;
-          if (shouldRetry) {
-            LOGGER.error("Retrying request on response status {}. Count {} Max is {}", statusCode, executionCount, MAX_RETRIES);
+          if (!shouldRetry) {
```

Comment:
Throwing an exception here because the the handler doesn't throw an exception explicitly when we aren't retrying a request.